server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -228,8 +228,9 @@ private void allocateUnassigned() {
                 }
 
                 final var allocationStatus = UnassignedInfo.AllocationStatus.fromDecision(
-                    isThrottled ? Decision.Type.THROTTLE : Decision.Type.NO
+                    isThrottled || desiredBalance.isBalanceComputed(shard.shardId()) == false ? Decision.Type.THROTTLE : Decision.Type.NO
```

Comment:
Hmm if we haven't computed the balance for a shard yet I would expect the right status to be `AllocationStatus.NO_ATTEMPT`. Does that not work?

Reply:
Unfortunately that still produces RED status